File: app/src/main/java/team/triplog/presentation/adapter/TripPlanAdapter.java
package team.triplog.presentation.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import team.triplog.R;

public class TripPlanAdapter extends RecyclerView.Adapter<TripPlanAdapter.ViewHolder> {
    private Context context;
    private ArrayList<String> plans;

    public TripPlanAdapter(Context context, ArrayList<String> plans) {
        this.context = context;
        this.plans = plans;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(
                LayoutInflater
                        .from(context)
                        .inflate(R.layout.item_trip_plan, parent, false)
        );
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        // TODO : entity 필드와 연결시키기

        String item = plans.get(position);
        holder.textLocation.setText(item);
    }

    @Override
    public int getItemCount() {
        return plans.size();
    }

    public void setItems(ArrayList<String> plans) {
        this.plans = plans;
        notifyDataSetChanged();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        View viewTop, viewBottom;
        TextView textLocation, textAddress;

        ViewHolder(View view) {
            super(view);

            textLocation = view.findViewById(R.id.text_location);
            textAddress = view.findViewById(R.id.text_address);
            viewTop = view.findViewById(R.id.view_top);
            viewBottom = view.findViewById(R.id.view_bottom);
        }
    }
}

File: app/src/main/java/team/triplog/presentation/adapter/TripPlanAdapter.kt
package team.triplog.presentation.adapter

import android.content.Context
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import team.triplog.presentation.viewholder.TripPlanViewHolder
import java.util.*

class TripPlanAdapter(
    private val context: Context,
    private var items: ArrayList<String>
) : RecyclerView.Adapter<TripPlanViewHolder>() {
    override fun getItemCount(): Int = items.size

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TripPlanViewHolder {
        return TripPlanViewHolder.getViewHolder(parent)
    }

    override fun onBindViewHolder(holder: TripPlanViewHolder, position: Int) {
        // TODO : entity 필드와 연결시키기

        val item = items[position]
        holder.bind(item)
    }

    fun setItems(items: ArrayList<String>) {
        this.items = items
        notifyDataSetChanged()
    }
}